### ISSUE TITLE
Allow the default type map to be overrided from config

### ISSUE THREAD

**Issue by adelsz** [adelsz/pgtyped#146]
Config should have a new `typeMap` field:
```js
{
  typeMap: {
    bigint: 'number' | 'bigint' | 'string',
    interval: 'iso-string' | 'interval' | 'string'
  }
}
```
This field allows to select a parser and type for each postgres type passed and returned.
pgTyped will then automatically handle parsing this type at both runtime and compile-time.

**Comment by logan12358** [comment]
This mapping can be achieved in node-postgres with the `pg.types.setTypeParser` mechanism. If I understand correctly, the `BigInt` mapping might become default for node-postgres in the future: https://github.com/brianc/node-pg-types/issues/78.

pgtyped implicitly relies on the type mappings of whichever `IDatabaseConnection` is used. Do you know if it's common to use backends other than node-postgres to provide `IDatabaseConnection`?

I think two big options are:
1. Support configuration over which types are emitted, but don't do runtime conversion. Add documentation for configuring backends to match.
2. Integrate more tightly with node-postgres and automatically set the client's type parser (following https://github.com/adelsz/pgtyped/milestone/1, I guess)

**Comment by adelsz** [comment]
Implemented by #476
